ranger/README.md

```diff
@@ -0,0 +1,29 @@
+# Apache Ranger
+
+This initialization action installs [Apache Ranger](https://ranger.apache.org/) on a [Google Cloud Dataproc](https://cloud.google.com/dataproc) cluster.
+ Apache Ranger enables monitoring and managing data security across the Hadoop ecosystem and uses [Apache Solr](http://lucene.apache.org/solr/) for audits.
+
+## Using this initialization action
+
+You can use this initialization action to create a new Dataproc cluster with Apache Ranger installed:
+
+1. Use the `gcloud` command to create a new cluster with this initialization action. 
+The following command will create a new high availability cluster named `<CLUSTER_NAME>`.
+
+    ```bash
+    gcloud dataproc clusters create <CLUSTER_NAME> \
+      --num-masters 3 \
+      --initialization-actions gs://dataproc-initialization-actions/solr/solr.sh,\
+    gs://dataproc-initialization-actions/ranger/ranger.sh \
+      --image-version 1.2
+    ```
+1. Once the cluster has been created Apache Ranger Policy Manager should be running on **m-0** and use Solr in SolrCloud mode for audits.
+1. The Policy Manager Web UI is served on port 6080. 
+Follow the instructions on [connect to cluster web interfaces](https://cloud.google.com/dataproc/docs/concepts/accessing/cluster-web-interfaces) 
+to create a SOCKS5 proxy to view `http://clustername-m-0:6080` in your browser.
+
+## Important notes
+* The default admin password is `dataproc2019` and can be changed in init action script. You can also change it in after the first log in.
+* Apache Ranger Policy Manager and usersync plugin are installed on master nodes only(m-0 in HA mode).
+* This script will install hdfs, hive and yarn plugin by default.
+* Ranger and whole plugin stack is supported only by Dataproc 1.2 and 1.1 at this moment.
```

Comment:
Can you mention the reason here?

Reply:
- default admin password will be configured by mandatory metadata 
- Ranger provides GUI which is called Ranger Policy Manager that has to be deployed on the main node in the cluster, user sync plugin allows to sync users installed on the clusters to Ranger so it could be done from any node in the cluster, but I just wanted to be consistent and also deployed this plugin to master nodes. Ranger doesn't support real HA, documentation mentions about setting up Apache web server that will act as the load balancer and split traffic, instead I just decided to install web just on m-0.
- those 3 plugins are being installed by default as they are available as a service on the cluster out of the box, so would be nice to handle them. We can change this behavior by adding metadata in which we can specify which additional plugins(like Kafka) should be activated installed, by this is sth bigger that could be done in separate PR
- I didn't mention dataproc 1.3 because yarn plugin is not working on dataproc 1.3. I wasn't able to successfully restart yarn resource manager after the installation of yarn plugin.